File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.stream.IntStream;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int M = scanner.nextInt();
        List<String> board = new ArrayList<String>();
        for (int i=0;i<N;i++){
            String word = scanner.next();
            board.add(word);
        }
        String checkW = "WBWBWBWB";
        String checkB = "BWBWBWBW";
        int minEraser = 99999999;
        for (int col=0;col<N-7;col++)
        {
            for (int row=0;row<M-7;row++)
            {
                int[] nowEraser = new int[2];
                for (int plus=0;plus<8;plus++)
                {
                    String check = board.get(col+plus).substring(row, row+8);
                    nowEraser[plus%2]+=getDiffCount(check, checkW);
                    nowEraser[1-plus%2]+=getDiffCount(check, checkB);
                    if (minEraser < Math.min(nowEraser[0],nowEraser[1])) {
                        break;
                    }
                }
                if (minEraser > Math.min(nowEraser[0],nowEraser[1])) {
                    minEraser = Math.min(nowEraser[0],nowEraser[1]);
                }
            }
        }
        System.out.println(minEraser);
    }
    public static int getDiffCount(String a, String b)
    {
        int diffCount = IntStream.range(0, Math.min(a.length(), b.length()))
            .map(i -> a.charAt(i) != b.charAt(i) ? 1 : 0)
            .sum() + Math.abs(a.length() - b.length());
        return diffCount;

    }
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
def getStartStr(start):
    if (start == "W"):
        return "WB"*4
    return "BW"*4
def diffStr(str_a,str_b):
    return sum(1 for a, b in zip(str_a, str_b) if a != b)
min_eraser = 99999
# broute - force 로 풀기
N,M = map(int,input().split())
board = [input() for _ in range(N)]
# 정사각형 8*8 찾는 반복문
for col in range(N-7):
    for row in range(M-7):
        start = board[col][row]
        start_char = ["W","B"]
        now_eraser = [0,0]
        for plus in range(8):
            check = board[col+plus][row:row+8]
            now_eraser[0] += diffStr(check,getStartStr(start_char[0]))
            now_eraser[1] += diffStr(check,getStartStr(start_char[1]))
            
            start_char = start_char[::-1]
            if (min_eraser < min(now_eraser)):
                break
        else:
            if (min_eraser > min(now_eraser)):
                min_eraser = min(now_eraser)
print(min_eraser)